RulesEngine/OatSubstringIndexClause.cs

```diff
@@ -0,0 +1,20 @@
+﻿// Copyright (C) Microsoft. All rights reserved. Licensed under the MIT License.
+
+using Microsoft.CST.OAT;
```

Comment:
Is OatSubstringIIndex generic enough to move into OAT: https://github.com/microsoft/OAT/tree/main/OAT/Operations?, or did I get that wrong?

Reply:
A few issues. 

It's not much different than the contains operator that already exists and is much more broad in terms of the data types it can work on. 

Secondly, built-in operations in OAT should not be for one specific data type (for example this just works for strings).

The main distinction is that we are capturing and returning the index of the match, a behavior that is not mirrored in any other OAT operation, and doesn't make contextual sense outside of simple string matches.

See the existing AI code and AI uses it's own regex instead of the OAT regex because OAT doesn't deal with Boundary objects, that's a AI concept.